doc/developer-notes.md

```diff
@@ -568,3 +568,16 @@ A few guidelines for introducing and reviewing new RPC interfaces:
   - *Rationale*: as well as complicating the implementation and interfering
     with the introduction of multi-wallet, wallet and non-wallet code should be
     separated to avoid introducing circular dependencies between code units.
+
+- Wallet RPCs call BlockUntilSyncedToCurrentChain to maintain consistency with
+  `getblockchaininfo`'s state immediately prior to the call's execution. Wallet
+  RPCs who's behavior does *not* depend on the current chainstate may omit this
```

Comment:
In commit "Add a dev notes document describing the new wallet RPC blocking"

s/who's/whose